src/modal/modal.js

```diff
@@ -390,14 +392,31 @@ angular.module('ui.bootstrap.modal', [])
 
           $modal.open = function (modalOptions) {
 
+            //merge and clean up options
+            modalOptions = angular.extend({}, $modalProvider.options, modalOptions);
+            modalOptions.resolve = modalOptions.resolve || {};
+
+            //verify options
+            if (!modalOptions.template && !modalOptions.templateUrl) {
+              throw new Error('One of template or templateUrl options is required.');
+            }
+
+            var templateAndResolvePromise =
+              $q.all([getTemplatePromise(modalOptions)].concat(getResolvePromises(modalOptions.resolve)));
+
             var modalResultDeferred = $q.defer();
-            var modalOpenedDeferred = $q.defer();
             var modalRenderDeferred = $q.defer();
 
+            //add opened, but ensure it resolves in order
+            var previous = modalOpenedPromiseChain;
+            modalOpenedPromiseChain = $q.all([previous || $q.when(undefined), templateAndResolvePromise])
+              .then(function() { return true; }, function(reason) { return reason[reason.length-1]; })
+              ['finally'](function() { modalOpenedPromiseChain = previous; });
```

Comment:
This `finally` may call prematurely if 3 modals are opened and the first modal's templateAndResolvePromise fails. In such a scenario, if a 4th modal is opened, the `modalOpenedPromiseChain` would be undefined, but that's not correct.